kyaml/yaml/rnode_test.go

```diff
@@ -721,6 +721,288 @@ data:
 `), strings.TrimSpace(actual))
 }
 
+func TestDeAnchorMerge(t *testing.T) {
+	testCases := []struct {
+		description string
+		input       string
+		expected    string
+	}{
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "simplest merge tag",
+			input: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color: &color-used
+    foo: bar
+  primaryColor:
+    <<: *color-used
+`,
+			expected: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color:
+    foo: bar
+  primaryColor:
+    foo: bar
+`,
+		},
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "keep duplicated keys",
```

Comment:
I'm curious what led you to the conclusion that this is important... it would be weird for this to be a point where we suddenly drop keys, but kyaml actually does not support duplicate keys in general--it inherits this behaviour from go-yaml, which takes that position because duplicates are not valid per the yaml spec. So if you try to simply roundtrip this test case with go-yaml, you'll get an error.

Reply:
I added this because other tests that check duplication of keys were failing (because the keys were not duplicate anymore). It just seems that this function shouldn't interfere with that check